### PR TITLE
ci: support `spell-checker`

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [x] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?

Best practice for big projects:

https://github.com/prettier/prettier/blob/main/cspell.json
https://github.com/swc-project/swc/blob/main/cspell.json
https://github.com/webpack/webpack/blob/main/cspell.json

### CODE REVIEW

File: .cspell.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        "**/i18n/README.md",
+        "**/projects/i18n/languages/**",
+        "**/cypress/support/editor/html.ts",
+        "**/ng-update/v3/steps/miscellaneous.ts",
+        "**/components/dialog/dialog.template.html",
+        "**/input-tag/input-tag.component.ts",
+        "**/components/line-clamp/line-clamp.template.html",
+        "**/tokens/is-mobile.ts",
+        "**/components/sheet/sheet.template.html",
+        "**/examples/**"
```
Comment (splincode) [inline .cspell.json:55]: I will fix it in the future

File: projects/core/observables/zonefull-map.ts [hunk 2]
```diff
@@ -10,9 +10,15 @@ import {distinctUntilChanged, map} from 'rxjs/operators';
  * @param project mapping function
  * @param ngZone NgZone instance
  */
-export function tuiZonefulMap<T, R>(
+export function tuiZonefullMap<T, R>(
```
Comment (waterplea) [inline projects/core/observables/zonefull-map.ts:13]: This is wrong. It's zoneful as in joyful.

File: projects/demo-integrations/cypress/support/helpers/wait-requests.util.ts [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
         .get<Array<{state: string}>>(`${alias}.all`, {log: false})
-        .then(reqs => reqs.filter(req => req.state !== `Complete`));
+        .then(request => request.filter(req => req.state !== `Complete`));
 
 export const waitAllRequests = (alias: string): void => {
     getNotLoadedRequests(alias)
-        .then(reqs => {
-            return reqs.length ? cy.wait(alias, {log: false}) : cy.tuiWaitBeforeAction();
+        .then(request => {
+            return request.length
```
Comment (waterplea) [inline projects/demo-integrations/cypress/support/helpers/wait-requests.util.ts:9]: Unnecessary return

File: projects/kit/components/input-number/input-number.component.ts [hunk 4]
```diff
@@ -120,7 +120,7 @@ export class TuiInputNumberComponent
 
     get inputMode(): TuiInputMode {
         if (this.isIOS && this.isNegativeAllowed) {
-            // iphones do not have minus sign if inputMode is equal to 'numeric' / 'decimal'
+            // iPhone do not have minus sign if inputMode is equal to 'numeric' / 'decimal'
```
Comment (waterplea) [inline projects/kit/components/input-number/input-number.component.ts:123]: Either iphoneS or "does"

File: .cspell.json (hunk 1 above)
Comment (splincode) [reply .cspell.json]: fixed

File: projects/demo/src/modules/app/app.browser.module.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ import {MetrikaModule} from 'ng-yandex-metrika';
 import {MarkdownModule} from 'ngx-markdown';
 
 import {environment} from '../../environments/environment';
-import {CustomHostModule} from '../customization/portals/examples/1/portal/сustom-host.module';
+import {CustomHostModule} from '../customization/portals/examples/1/portal/custom-host.module';
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/app/app.browser.module.ts:21]: fixed

File: projects/cdk/schematics/ng-update/v3/constants/constants.ts [hunk 6]
```diff
@@ -152,7 +152,7 @@ export const CONSTS_TO_REPLACE: ReplacementConst[] = [
     },
     {
         from: {
-            name: `TableComands`,
+            name: `TableCommands`,
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/schematics/ng-update/v3/constants/constants.ts]: fixed

File: projects/demo/src/modules/components/input-copy/input-copy.template.html [hunk 7]
```diff
@@ -4,7 +4,7 @@
     type="components"
 >
     <ng-template pageTab>
-        <p>InputCopy alows user to copy a text of textfield</p>
+        <p>InputCopy allow user to copy a text of textfield</p>
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/input-copy/input-copy.template.html]: fixed

**Comment by splincode** [comment]
@vladimirpotekhin @nsbarsukov hello, please review me)

**Comment by splincode** [comment]
@nsbarsukov hello, please review me

File: projects/core/observables/index.ts [hunk 8]
```diff
@@ -1,2 +1,2 @@
 export * from './smart-search';
-export * from './zoneful-map';
+export * from './zonefull-map';
```
Comment (nsbarsukov) [inline projects/core/observables/index.ts:2]: Should we also revert filename ?

https://github.com/Tinkoff/taiga-ui/pull/4292#discussion_r1175277003

**Review by nsbarsukov — APPROVED** [review]

File: projects/core/observables/index.ts (hunk 8 above)
Comment (splincode) [reply projects/core/observables/index.ts:2]: oh, sorry

File: projects/demo/src/modules/components/avatar/examples/1/index.html [hunk 9]
```diff
@@ -3,11 +3,11 @@
     [avatarUrl]="avatar"
 ></tui-avatar>
 <tui-avatar
-    text="alex inkin"
```
Comment (waterplea) [inline projects/demo/src/modules/components/avatar/examples/1/index.html:6]: It was written this way to show that it uppercases the first letters :)

File: projects/eslint-plugin/rules/no-assert-without-ng-dev-mode.js [hunk 10]
```diff
@@ -29,7 +29,7 @@ module.exports = {
                         items: {
                             type: 'string',
                         },
-                        description: 'An array of callees that will be checked.',
+                        description: 'An array of callers that will be checked.',
```
Comment (waterplea) [inline projects/eslint-plugin/rules/no-assert-without-ng-dev-mode.js:32]: Cellee is a valid word:
https://en.wiktionary.org/wiki/callee
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee

**Pull request merged**